apps/pkeyutl.c

```diff
@@ -295,14 +286,33 @@ int pkeyutl_main(int argc, char **argv)
         goto opthelp;
     }
 
+    pkey = get_pkey(kdfalg, inkey, keyform, key_type, passinarg, pkey_op, e);
+    if (pkey_op == EVP_PKEY_OP_SIGN || pkey_op == EVP_PKEY_OP_VERIFY) {
+        if (only_rawin(pkey))
+            rawin = 1; /* implied for Ed25519 and Ed448 */
+    } else if (rawin) {
+        BIO_printf(bio_err,
+                   "%s: -rawin can only be used with -sign or -verify\n", prog);
+        goto opthelp;
+    }
+    if (digestname != NULL && !rawin) {
+        BIO_printf(bio_err,
+                   "%s: -digest can only be used with -rawin\n", prog);
+        goto opthelp;
+    }
+
+    if (rawin && rev) {
+        BIO_printf(bio_err, "%s: -rev cannot be used with raw input\n", prog);
```

Comment:
How is pkey freed in this case?

Reply:
`pkey` gets freed as in the usual way, 
as can be seen easily following the `opthelp` and then `end` labels.

Reply:
Oops, I had some confusion; will fix the mem leak...